pkg/services/ngalert/provisioning/contactpoints.go

```diff
@@ -34,17 +34,26 @@ func NewContactPointService(store AMConfigStore, encryptionService secrets.Servi
 	}
 }
 
-func (ecp *ContactPointService) GetContactPoints(ctx context.Context, orgID int64) ([]apimodels.EmbeddedContactPoint, error) {
-	revision, err := getLastConfiguration(ctx, orgID, ecp.amStore)
+type ContactPointQuery struct {
+	Name  string
```

Comment:
We should document that this is optional.

Reply:
Done

I had a thought along these lines - perhaps we should make orgID a separate parameter to GetContactPoints and not put in the query? It's definitely required and it'd be more type safe to force the caller to deal with the parameter. Go doesn't force you to fill in any struct fields, so someone could miss it.

Reply:
I'm a little torn on this. I like the simplicity of `GetContactPoints(context, query)` too, it's rather clean.

Reply:
Went with just the documentation for now for the sake of simplicity. We can always extend this.